### PR TITLE
Adopt potentials to predict user defined number of scalar properties 

### PR DESCRIPTION
**Pull request by wiederm**
## Description

This PR will modify each model to output multiple scalar properties upon request using different output heads (as a general strategy, some models have their own strategy that will be preserved if present).

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [x] update TOML format
  - [x] update Pydantic classes
  - [x] Update SchNet
  - [x] Update PhysNet
  - [x] Update ANI2x
  - [x] Update TensorNEt
  - [x] Update PaiNN
  - [x] Update SAKE

## Questions
- [x] Question1

## Status
- [x] Ready to go

### CODE REVIEW

File: modelforge/potential/sake.py [hunk 1]
```diff
@@ -137,6 +131,27 @@ def __init__(
             for _ in range(self.nr_interaction_blocks)
         )
 
+        # Initialize output layers based on configuration
+        self.output_layers = nn.ModuleDict()
+        for property in predicted_properties:
+            output_name = property["name"]
+            output_type = property["type"]
+            output_dimension = (
+                1 if output_type == "scalar" else 3
```
[the remark this answers is not in the document]
Comment (wiederm) [reply modelforge/potential/sake.py:140]: That is a good point! I will avoid hard-coding here too!

File: modelforge/potential/sake.py [hunk 2]
```diff
@@ -192,14 +207,18 @@ def compute_properties(self, data: SAKENeuralNetworkInput):
...
-
-        return {
-            "per_atom_energy": E_i,
+        results = {
+            "per_atom_scalar_representation": h,
             "atomic_subsystem_indices": data.atomic_subsystem_indices,
         }
 
+        # Compute all specified outputs
+        for output_name, output_layer in self.output_layers.items():
```
[the remark this answers is not in the document]
Comment (wiederm) [reply modelforge/potential/sake.py:216]: I don't know if there is a definitive best approach to this.

Having multiple output heads is an advantage when the magnitude or mean of the properties you predict differ. But it is a bit more involved to implement and there is an additional cost associated with the fact that this is currently implemented as a loop.

Changing this to a single MLP with multiple output dimensions is trivial though.
